File: 20190205_Android/HelloAndroid/app/src/main/java/com/example/helloandroid/MainActivity.java
package com.example.helloandroid;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    // 버튼이 클릭된 횟수를 저장하는 변수를 선언한다.
    int clickCount = 0;

    /**
     * MainActivity 가 최초로 실행될 때 실행된다.
     * 자바독 주석을 확인하려면 F1 키를 누르자.
     *
     * @param savedInstanceState
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Activity 의 UI 를 R.layout.activity_main 으로 지정한다.
        setContentView(R.layout.activity_main);

        // "프로그래밍을 시작해보자" 메세지를 잠시 보여준다.
        Toast.makeText(getApplicationContext(), "프로그래밍을 시작해보자", Toast.LENGTH_LONG).show();
        /*
        여러 주석을 사용하는 방법
        주석2줄
        주석3줄
         */

        // 레이아웃에 button 이라는 ID 로 선언된 뷰에 클릭 이벤트 리스너를 등록한다.
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // 클릭 카운트를 1회 증가 시킨다.
                clickCount = clickCount + 1;

                // 화면에 보여줄 메세지를 저장할 변수를 선언한다.
                String text = "";

                // 클릭한 횟수만큼 반복하면서 메세지를 만든다.
                for (int i = 0; i < clickCount ; i++) {
                    if(i >= 3){
                        break;
                    }
                    text = text + "안녕.";
                }


                Toast.makeText(getApplicationContext(), text, Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: 20190205_Android/MyKotlin/app/src/main/java/com/example/mykotlin/MainActivity.kt
package com.example.mykotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // button1 클릭된 경우의 이벤트 리스너 설정
        button1.setOnClickListener {
            // Intent 로 BmiJavaActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent   (this@MainActivity, BmiJavaActivity::class.java))
        }

        button2.setOnClickListener{
            // Intent로 BmiKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent (this@MainActivity, BmiKotlinActivity::class.java))
        }
        button3.setOnClickListener{
            // Intent로 VariableJavaActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent (this@MainActivity, VariableJavaActivity::class.java))
        }
        button4.setOnClickListener{
            // Intent로 VariableKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent (this@MainActivity, VariableKotlinActivity::class.java))
        }
        button5.setOnClickListener{
            // Intent로 VariableKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent (this@MainActivity, ControlJavaActivity::class.java))
        }
        button6.setOnClickListener{
            // Intent로 VariableKotlinActivity를 타겟으로 지정하고 startActivity로 실행
            startActivity(Intent (this@MainActivity, ControlKotlinActivity::class.java))
        }
    }
}
